app/models/machine.rb

```diff
@@ -2,4 +2,10 @@ class Machine < ApplicationRecord
   validates_presence_of :location
   
   belongs_to :owner
+  has_many :machine_snacks
+  has_many :snacks, through: :machine_snacks
+
+  def average_snack_price
+    snacks.average('price::float').to_s
+  end
```

Comment:
did you try something like what i wrote below for a class method on the snack model (referring to your comment)?
```ruby
def self.average_price
   average('price::float').to_s
end 
```